includes/Core/Authentication/Clients/OAuth_Client.php

```diff
@@ -404,6 +416,19 @@ function ( $scope ) {
 	public function authorize_user() {
 		$code       = htmlspecialchars( $this->context->input()->filter( INPUT_GET, 'code' ) ?? '' );
 		$error_code = htmlspecialchars( $this->context->input()->filter( INPUT_GET, 'error' ) ?? '' );
+
+		// If we have a code, check if there's a stored redirect URL to prevent duplicate setups.
```

Comment:
Hi @hussain-t, it would be useful to provide a bit more detail as to why it would be problematic to continue again with the same auth code.

I am assuming it's because `fetchAccessTokenWithAuthCode( $code )` should only be called once with a given code, however, trying to do a quick test on `develop` I'm no longer able to reproduce the error to validate this, maybe due to the service-side fix @eugene-manuilov mentioned [here](https://github.com/google/site-kit-wp/issues/10037#issuecomment-2619255482).

Can you provide a bit more detail in the comments?

Reply:
Sure, I'll provide more details in the comments. Btw, I can reproduce the issue in the latest `develop`.

Reply:
Updated.

Reply:
Thanks @hussain-t!

Interesting to know you can still reproduce the issue on `develop`, I wonder why I can't. Anyhow, it's not worth digging too deeply into now, things are working well for me with this issue and you've been able to validate it vs `develop`. All SGTM.